linux/ubuntu/scripts/helpers/install.sh

```diff
@@ -3,44 +3,52 @@
 ##  File:  install.sh
 ##  Desc:  Helper functions for installing tools
 ################################################################################
+# source: https://github.com/actions/runner-images/blob/5d6938f680075d63fa71f8aa70990866cd12884b/images/linux/scripts/helpers/install.sh
 
 download_with_retries() {
-  # Due to restrictions of bash functions, positional arguments are used here.
-  # In case if you using latest argument NAME, you should also set value to all previous parameters.
-  # Example: download_with_retries $ANDROID_SDK_URL "." "android_sdk.zip"
-  local URL="$1"
-  local DEST="${2:-.}"
-  local NAME="${3:-${URL##*/}}"
-  local COMPRESSED="${4:-}"
-
-  if [[ $COMPRESSED == "compressed" ]]; then
-    local COMMAND="curl $URL -4 -sL --compressed -o '$DEST/$NAME' -w '%{http_code}'"
-  else
-    local COMMAND="curl $URL -4 -sL -o '$DEST/$NAME' -w '%{http_code}'"
-  fi
-
-  echo "Downloading '$URL' to '${DEST}/${NAME}'..."
-  retries=20
-  interval=30
-  while [ $retries -gt 0 ]; do
-    ((retries--))
-    # Temporary disable exit on error to retry on non-zero exit code
-    set +e
-    http_code=$(eval "$COMMAND")
-    exit_code=$?
-    if [ "$http_code" -eq 200 ] && [ $exit_code -eq 0 ]; then
-      echo "Download completed"
-      return 0
+# Due to restrictions of bash functions, positional arguments are used here.
+# In case if you using latest argument NAME, you should also set value to all previous parameters.
+# Example: download_with_retries $ANDROID_SDK_URL "." "android_sdk.zip"
+    local URL="$1"
+    local DEST="${2:-.}"
+    local NAME="${3:-${URL##*/}}"
+    local COMPRESSED="$4"
+
+    if [[ $COMPRESSED == "compressed" ]]; then
+        local COMMAND="curl $URL -4 -sL --compressed -o '$DEST/$NAME' -w '%{http_code}'"
     else
-      echo "Error — Either HTTP response code for '$URL' is wrong - '$http_code' or exit code is not 0 - '$exit_code'. Waiting $interval seconds before the next attempt, $retries attempts left"
-      sleep 30
+        local COMMAND="curl $URL -4 -sL -o '$DEST/$NAME' -w '%{http_code}'"
     fi
-    # Enable exit on error back
-    set -e
-  done
 
-  echo "Could not download $URL"
-  return 1
+    # Save current errexit state and disable it to prevent unexpected exit on error
+    if echo $SHELLOPTS | grep '\(^\|:\)errexit\(:\|$\)' > /dev/null;
+    then
+        local ERR_EXIT_ENABLED=true
+    else
+        local ERR_EXIT_ENABLED=false
+    fi
+    set +e
```

Comment:
The changes to this script aren't my own. They're from the upstream repository.

Primary reason I found the sync was necessary was because checksum validation code was added and used for the `yq` download.

I'll have to defer to the blame upstream for your questions though :]

Here it's https://github.com/actions/runner-images/pull/8352

Reply:
Just to be clear, I'm not defending the decisions upstream. I don't have an opinion on the suggestions you made here.

But I think there's tradeoffs maintenance between how closely you want to track the other repo. So I feel the decision to diverge and track our own patches here should be up to the maintainers and not me :]

Feel free to edit, or request changes.